packages/contentstack-import/src/import/modules/global-fields.ts

```diff
@@ -85,46 +91,80 @@ export default class ImportGlobalFields extends BaseClass {
-      fsUtil.writeFile(this.gFsUidMapperPath, this.gFsUidMapper);
-      log(this.config, 'Global field ' + uid + ' created successfully', 'success');
+      log(this.importConfig, `Global field ${globalField.uid} created successfully`, 'success');
     };
     const onReject = ({ error, apiData: globalField = undefined }: any) => {
-      const uid = globalField.uid;
+      const uid = globalField?.uid;
       if (error?.errors?.title) {
         if (this.importConfig.replaceExisting) {
           this.existingGFs.push(globalField);
         }
         if (!this.importConfig.skipExisting) {
-          log(this.importConfig, `Global fields '${uid}' already exist`, 'info');
+          log(this.importConfig, `Global fields '${globalField?.global_field?.uid}' already exist`, 'info');
         }
       } else {
         log(this.importConfig, `Global fields '${uid}' failed to import`, 'error');
         log(this.importConfig, formatError(error), 'error');
         this.failedGFs.push({ uid });
       }
     };
+    return await this.makeConcurrentCall({
+      processName: 'Import global fields',
+      apiContent: this.gFs,
+      apiParams: {
+        serializeData: this.serializeGFs.bind(this),
+        reject: onReject.bind(this),
+        resolve: onSuccess.bind(this),
+        entity: 'create-gfs',
+        includeParamOnCompletion: true,
+      },
+      concurrencyLimit: this.reqConcurrency,
+    });
+  }
 
-    return await this.makeConcurrentCall(
-      {
-        processName: 'Import global fields',
-        apiContent: this.gFs,
-        apiParams: {
-          reject: onReject.bind(this),
-          resolve: onSuccess.bind(this),
-          entity: 'create-gfs',
-          includeParamOnCompletion: true,
-        },
-        concurrencyLimit: this.reqConcurrency,
+  /**
+   * @method serializeGFs
+   * @param {ApiOptions} apiOptions ApiOptions
+   * @returns {ApiOptions} ApiOptions
+   */
+  serializeGFs(apiOptions: ApiOptions): ApiOptions {
+    const { apiData: globalField } = apiOptions;
+    const updatedGF = cloneDeep(gfSchemaTemplate);
+    updatedGF.global_field.uid = globalField.uid;
+    updatedGF.global_field.title = globalField.title;
+    apiOptions.apiData = updatedGF;
+    return apiOptions;
+  }
+
+  async updateGFs(): Promise<any> {
+    const onSuccess = ({ response: globalField, apiData: { uid } = undefined }: any) => {
+      log(this.importConfig, `Updated the global field ${uid}`, 'info');
+    };
+    const onReject = ({ error, apiData: { uid } = undefined }: any) => {
+      log(this.importConfig, `Failed to update the global field '${uid}' ${formatError(error)}`, 'error');
+    };
+    
+    return await this.makeConcurrentCall({
+      processName: 'Update Global Fields',
+      apiContent: this.gFs,
+      apiParams: {
+        serializeData: this.serializeUpdateGFs.bind(this),
+        reject: onReject.bind(this),
+        resolve: onSuccess.bind(this),
+        entity: 'update-gfs',
```

Comment:
Why do we still need to have serialize data function and an entity here ? since we have a promisifying handler